src/vscode.rs

```diff
@@ -0,0 +1,86 @@
+use fs4::tokio::AsyncFileExt;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::collections::HashMap;
+use tokio::{
+    fs::OpenOptions,
+    io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt},
+};
+
+use crate::dirs::vscode_user_settings_file_path;
+
+const AQORA_CAN_INSTALL_EXT_KEY: &str = "aqora.canInstallExtensions";
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct UserVSCodeSettings {
+    #[serde(flatten)]
+    pub other: HashMap<String, serde_json::Value>,
+}
+
+// Sometimes, users may provide JSON with a trailing comma,
+// which serde does not support.
+fn clean_json(input: impl AsRef<str>) -> String {
+    let re = Regex::new(r",\s*(\}|\])").unwrap();
+    re.replace_all(input.as_ref(), "$1").to_string()
+}
+
+impl UserVSCodeSettings {
+    pub async fn load() -> Result<Self, std::io::Error> {
+        let path = vscode_user_settings_file_path();
+        let mut file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .truncate(false)
+            .open(&path)
+            .await?;
+
+        let _ = file.lock_exclusive();
```

Comment:
I think you have to make sure that this lock drops if any errors occurs after it because it doesn't unlock automatically. so you'll have to wrap the statements below

Reply:
Agreed